learnupon_exporter/management/commands/learnupon_export_enrollment_data.py

```diff
@@ -22,117 +22,88 @@ class Command(ExportCommand):
-                self.logger.info('Fetching %s of %s', index + 1, user_count)
-            user_enrollment_ids = self.get_enrollment_ids_for_user(user, course_mappings)
-            yield from user_enrollment_ids.items()
-
     def export_enrollment_data(self, csv_file):
         """
         Export all enrollment data in the system
         """
         field_mapping = {
             "Login ID": "user.email",
-            "Course Name": "course_id",
-            "Enrollment Status": "",
-            "Enrollment Completed Date": "",
-            "Enrollment Created Date": "",
+            "Course Name": "course.display_name",
+            "Enrollment Created Date": "created",
             "Enrollment Started Date": "",
-            "Enrollment Score": "",
-            "Enrollment Access Expires Date": ""
+            "Enrollment Completed Date": "",
+            "Enrollment Status": "",
+            "Enrollment Access Expires Date": "course.end",
         }
 
         writer = csv.DictWriter(csv_file, fieldnames=field_mapping)
 
         writer.writeheader()
-        enrollment_statuses = {es.learndot_enrolment_id: es for es in EnrolmentStatusLog.objects.all()}
+        enrollments = self.enrollments
+
+        record_count = self.enrollments.count()
+        self.stdout.write(f"Writing {record_count} enrollments")
+        batch_size = int(record_count / 10) or 1
+
+        for record_index, enrollment in enumerate(enrollments):
+            if record_index > 0 and record_index % batch_size == 0:
+                self.stdout.write(f"Completed: {record_index + 1} of {record_count}")
 
-        for enrollment, enrollment_id in self.fetch_learndot_enrollments():
             pseudo_context = {"obj": enrollment}
             row = {}
             for csv_field, instance_field in field_mapping.items():
                 if not instance_field:
-                    row[csv_field] = ""
-                else:
-                    row[csv_field] = Variable(f"obj.{instance_field}").resolve(pseudo_context)
-
-            row['Enrollment Status'] = 'not started'
-            enrollment_status = enrollment_statuses.get(enrollment_id)
-            if enrollment_status:
-                grade = enrollment_status.status.lower()
-                if grade == 'passed':
-                    row['Enrollment Status'] = 'passed'
-                    row['Enrollment Completed Date'] = self.format_date(enrollment_status.updated_at)
+                    continue
+                row[csv_field] = Variable(f"obj.{instance_field}").resolve(
+                    pseudo_context
+                )
+
+            row["Enrollment Status"] = "not started"
+            row["Enrollment Started Date"] = enrollment.created
+            modules = enrollment.user.studentmodule_set.filter(
+                course_id=enrollment.course_id
+            ).order_by("created")
+            first_block_viewed_at = modules.values_list("created", flat=True).first()
```

Comment:
I guess you could say "enrollment completed at" `modules.values_list("created", flat=True).last()` instead of `first()`? That way the completed date is as late as possible in the process..